### PR TITLE
Add q4z keyboard

### PR DESCRIPTION
**Pull request by rjboone**
## Description

This pull request contains the necessary files to add Q4Z to QMK main. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Initial addition of keyboard

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/q4z/config.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#pragma once
```
Comment (drashna) [inline keyboards/q4z/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/q4z/info.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (drashna) [inline keyboards/q4z/info.json:1]: This file should be filled out or removed. 

Ideally, filled out.  If you have the kbfirmware json, you can use this: https://noroadsleft.github.io/kbf_qmk_converter/

File: keyboards/q4z/keymaps/default/config.h [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#pragma once
```
Comment (drashna) [inline keyboards/q4z/keymaps/default/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/q4z/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,86 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/q4z/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/q4z/keymaps/rjboone/keymap.c [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  [COMBO_DEL] = COMBO(combo_del,KC_DEL),
+
+};
+#endif
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+
+#ifdef COMBO_ENABLE
+    //switch(get_highest_layer(state)) {
+    switch(get_highest_layer(state|default_layer_state)) {
```
Comment (drashna) [inline keyboards/q4z/keymaps/rjboone/keymap.c:97]:
```suggestion
    switch(get_highest_layer(state)) {
```

File: keyboards/q4z/q4z.c [hunk 6]
```diff
@@ -0,0 +1 @@
+#include "q4z.h"
```
Comment (drashna) [inline keyboards/q4z/q4z.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/q4z/q4z.h [hunk 7]
```diff
@@ -0,0 +1,19 @@
+#pragma once
```
Comment (drashna) [inline keyboards/q4z/q4z.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/q4z/q4z.h [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  K02, K07, K12, K17, K22, K26, K30, K34, K38,  K41, \
+  K03,  K08, K13, K18, K23, K27, K31, K35, K39, K42, \
+  K04,    K09, K14, K19, K24, K28, K32, K36,    K40, \
+  K05, K10, K15,     K20,    K25,    K29, K33, K37 \
+) { \
+{ K01, K06, K11, K16, K21 }, \
+{ K02, K07, K12, K17, K22, K26, K30, K34, K38, K41 }, \
+{ K03, K08, K13, K18, K23, K27, K31, K35, K39, K42 }, \
+{ K04, K09, K14, K19, K24, K28, K32, K36, K40 }, \
+{ K05, K10, K15, K20, K25, K29, K33, K37 } \
```
Comment (fauxpark) [inline keyboards/q4z/q4z.h:33]:
```suggestion
    K01, K06, K11, K16, K21, \
    K02, K07, K12, K17, K22, K26, K30, K34, K38,  K41, \
    K03,  K08, K13, K18, K23, K27, K31, K35, K39, K42, \
    K04,    K09, K14, K19, K24, K28, K32, K36,    K40, \
    K05, K10, K15,     K20,    K25,    K29, K33, K37 \
) { \
    { K01, K06, K11, K16, K21, XXX, XXX, XXX, XXX, XXX }, \
    { K02, K07, K12, K17, K22, K26, K30, K34, K38, K41 }, \
    { K03, K08, K13, K18, K23, K27, K31, K35, K39, K42 }, \
    { K04, K09, K14, K19, K24, K28, K32, K36, K40, XXX }, \
    { K05, K10, K15, K20, K25, K29, K33, K37, XXX, XXX } \
```

File: keyboards/q4z/readme.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
+# Q4Z
```
Comment (fauxpark) [inline keyboards/q4z/readme.md:1]: This file should follow the [template](https://github.com/qmk/qmk_firmware/blob/master/quantum/template/avr/readme.md).

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/q4z/readme.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+* Hardware Supported: Pro Micro or equivalent
+* Hardware Availability: This board is available to purchase from [P3D Store](https://p3dstore.com)
+
+Make example for this keyboard (after setting up your build environment):
+
+`make q4z:default`
+
+Flashing example for this keyboard:
+
+`make q4z:default:flash`
```
Comment (fauxpark) [inline keyboards/q4z/readme.md:18]:
```suggestion
    make q4z:default

Flashing example for this keyboard:

    make q4z:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**